### PR TITLE
chap: migrate from python-crypt to python-cryptography

### PR DESCRIPTION
**Pull request by rjfd**
Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

File: ceph_iscsi_config/client.py [hunk 1]
```diff
@@ -679,10 +680,15 @@ def _decrypt(self):
...
         try:
-            key = RSA.importKey(open(key_path))
-
-            cipher = PKCS1_OAEP.new(key)
-            plain_pw = cipher.decrypt(b64decode(self.password_str))
+            with open(key_path, 'rb') as keyf:
+                key = serialization.load_pem_private_key(keyf.read(), None,
+                                                         default_backend())
+
+            plain_pw = key.decrypt(b64decode(self.password_str),
```
Comment (dillaman) [inline ceph_iscsi_config/client.py]: Nit: I might be wrong, but I think the older library might have defaulted to the insecure SHA-1 hash. If the CHAP password was previously encrypted w/ the SHA-1 hash, would it fail to decode here? If so, for initial upgrade backwards compatibility, should it catch ```ValueError``` and try again w/ the downgrade?

Comment (rjfd) [reply ceph_iscsi_config/client.py]: I think you might be right. I forgot about the upgrade path. I'll add a unit test that encrypts the password using the previous library (python-crypto) and decrypt with the new code. That will give us some confidence.

**Comment by rjfd** [comment]
@dillaman the code can now handle the case where the password had been encrypted with the previous code

**Comment by rjfd** [comment]
looks like jenkins slave does not have python3-devel package installed, and pip is unable to build pycrypto for python 3 environment.

**Comment by LenzGr** [comment]
FYI: this Fixes github issue #6 (would it make sense that ceph-iscsi issues are tracked on the Ceph bug tracker as well?)

**Comment by dillaman** [comment]
@rjfd Perhaps just re-use python-cryptography but encrypt it in your test using:

```
            encrypted_pw = b64encode(key.encrypt(self.password_str.encode('utf-8'),
                                     padding.OAEP(
                                         mgf=padding.MGF1(algorithm=hashes.SHA1()),
                                         algorithm=hashes.SHA1(),
                                         label=None))).decode('utf-8')

```

**Comment by rjfd** [comment]
@dillaman done

**Pull request merged**